lua/better-breachable-doors/sv_breach.lua

```diff
@@ -0,0 +1,757 @@
+
+            -- Start checking to see if the player leaves the door's space
+            BBD.NonSolidDoors[ door ] = true
+        else
+            door:SetCollisionGroup( COLLISION_GROUP_NONE )
+            BBD.PlayRespawnSound( door )
+        end
+    else
+        BBD.PlayRespawnSound( door )
+    end
+
+    door:NextThink( CurTime() )
+end
+
+--#endregion Door Respawn Logic
+
+--#region Door Breach Logic
+
+-- Breaches a door by opening it violently
+---@param door BBD.Door Door to breach
+---@param dmg CTakeDamageInfo Damage that breached the door
+BBD.OpenBreachDoor = function( door, dmg )
+
+    -- Figure out which direction the damage is pushing the door
+    local openDirection = BBD.GetDoorOpenDirection( door, dmg )
+    door:SetDamageDirection( openDirection )
+
+    -- Get the door's normal open angle in the direction it's being pushed
+    local goalAng = door:GetInternalVariable( openDirection == BBD.OPEN_DIRECTION_FORWARD and "m_angRotationOpenForward" or "m_angRotationOpenBack" )
+
+    -- Modify the normal open angle to make the door look broken after it's opened
+    goalAng = goalAng + Vector( 0,0, -BBD.BreachedDoorTiltAmount )
+
+    -- Use the correct open speed for the damage type
+    local openSpeed = dmg:IsExplosionDamage() and conVarExplosiveSpeed:GetFloat() or conVarOpenSpeed:GetFloat()
+
+    -- Silence the door's opening sound
+    local previousSpawnFlags = door:GetSpawnFlags()
+    local silentFlags = bit.bor( previousSpawnFlags, 4096 ) -- 4096 is DOOR_FLAG_SILENT
+    door:SetKeyValue( "spawnflags", silentFlags )
+
+    -- Open the door to set its MoveDone function
+    door:Input( "Open" )
+
+    -- Override the breached door's movement to breach it open
+    AngularMove( door, goalAng, openSpeed )
+
+    -- Reset the door's spawnflags to their previous values
+    door:SetKeyValue( "spawnflags", previousSpawnFlags )
+
+    BBD.PlayBreachSound( door )
+end
+
+-- Breaches a door by replacing it with a prop door and throwing the prop door inward
+---@param door BBD.Door The door to replace with a prop door.
+---@param dmg CTakeDamageInfo Damage that killed the door.
+BBD.PropBreachDoor = function( door, dmg )
+
+    -- Silence the door's opening sound
+    local previousSpawnFlags = door:GetSpawnFlags()
+    local silentFlags = bit.bor( previousSpawnFlags, 4096 ) -- 4096 is DOOR_FLAG_SILENT
+    door:SetKeyValue( "spawnflags", silentFlags )
+
+    -- Open the door to trigger area portals
+    door:Input( "Open" )
+
+    -- Reset the door's spawnflags to their previous values
+    door:SetKeyValue( "spawnflags", previousSpawnFlags )
+
+    door:SetSolid( SOLID_NONE )
+
+    local prop = ents.Create( "prop_physics" )
+    prop:SetModel( door:GetModel() )
+    prop:SetPos( door:GetPos() )
+    prop:SetAngles( door:GetAngles() )
+
+    -- Done in a timer because props don't exist on the client until the next tick
+    timer.Simple( 0, function() door:SetPropDoor( prop ) end )
```

Comment:
You may want to include an IsValid check here. darkrp slop servers will do something dumb that deletes your new `prop` in this tick, and your addon will error through no fault of your own